packages/fxa-auth-client/lib/crypto.ts

```diff
@@ -206,6 +206,39 @@ export async function jweEncrypt(
   return compactJWE;
 }
 
+/**
+ * Decrypt a JWE token using provided key material
+ * @param keyMaterial Key material
+ * @param jwe JWE token
+ */
+export async function jweDecrypt(keyMaterial: Uint8Array, jwe: string) : Promise<string> {
+ const encoder = new TextEncoder();
+  const [header, empty, iv, ciphertext, authenticationTag] = jwe.split('.');
+  const key = await crypto.subtle.importKey(
+    'raw',
+    keyMaterial,
+    {
+      name: 'AES-GCM',
+    },
+    false,
+    ['decrypt']
+  );
+  const ciphertextBuf = base64UrlToUint8(ciphertext);
+  const authenticationTagBuf = base64UrlToUint8(authenticationTag);
+  const dataBuf = new Uint8Array([...ciphertextBuf, ...authenticationTagBuf]);
+  const decrypted = await crypto.subtle.decrypt(
+    {
+      name: 'AES-GCM',
+      iv: base64UrlToUint8(iv),
+      additionalData: encoder.encode(header),
+      tagLength: 128,
+    },
+    key,
+   dataBuf
+  );
+  const decoder = new TextDecoder();
+  return decoder.decode(decrypted);
+}
```

Comment:
Any reason you don't want to use node-jose for this?

Reply:
@chenba After going through this PR, I think we can actually ditch node-jose and use web crypto for everything.